### PR TITLE
Obtain latest head from head broadcaster

### PR DESCRIPTION
**Pull request by makramkd**
Ticket: https://app.shortcut.com/chainlinklabs/story/21440/use-the-headbroadcaster-like-v1-vrf-for-latest-head

Use a waitgroup to wait on running goroutines to finish rather than multiple waits on the same channel.

### CODE REVIEW

File: core/services/vrf/delegate.go [hunk 1]
```diff
@@ -12,14 +13,15 @@ import (
 
 	"github.com/ethereum/go-ethereum/common"
 	"github.com/pkg/errors"
+	"gorm.io/gorm"
```
[the remark this answers is not in the document]
Comment (makramkd) [reply core/services/vrf/delegate.go:16]: This is branched off a pretty old commit, hence the presence of gorm nearly everywhere. The commit we branched off is the one of the currently deployed VRF node, although it looks like we should be able to merge into develop at this point. I'm going to rebase vrfv2-fixes on develop and that should remove all the gorm.

File: core/services/vrf/listener_v2.go [hunk 2]
```diff
@@ -109,24 +117,63 @@ func (lsn *listenerV2) Start() error {
...
+func (lsn *listenerV2) setLatestHead(head eth.Head) {
+	lsn.latestHeadMu.Lock()
+	defer lsn.latestHeadMu.Unlock()
+	num := uint64(head.Number)
+	if num > lsn.latestHeadNumber {
+		lsn.latestHeadNumber = num
+	}
+}
+
+func (lsn *listenerV2) runHeadListener(unsubscribe func(), wg *sync.WaitGroup) {
```
Comment (connorwstein) [inline core/services/vrf/listener_v2.go:159]: hm I don't think we need this goroutine? We can just pass the head broadcaster unsubscribe into the list of unsubscribes arg in the runLogListener

Comment (makramkd) [reply core/services/vrf/listener_v2.go:159]: You're right, it's literally just for the unsubscribe. Will remove it.

**Comment by cl-homu** [comment]
:umbrella: The latest upstream changes (presumably 505b11953d9c1daa5c0307560219028a0fa098a4) made this pull request unmergeable. Please [resolve the merge conflicts](https://rustc-dev-guide.rust-lang.org/git.html#rebasing-and-conflicts).

**Review by connorwstein — APPROVED** [review]

**Pull request merged**